src/udev/udev-builtin-net_id.c

```diff
@@ -566,6 +566,51 @@ static int pci_get_hotplug_slot(sd_device *dev, uint32_t *ret) {
         return -ENOENT;
 }
 
+static int get_device_firmware_node_sun(sd_device *dev, uint32_t *ret) {
+        const char *attr;
+        int r;
+
+        assert(dev);
+        assert(ret);
+
+        r = device_get_sysattr_value_filtered(dev, "firmware_node/sun", &attr);
+        if (r < 0)
+                return log_device_debug_errno(dev, r, "Failed to read firmware_node/sun, ignoring: %m");
+
+        r = safe_atou32(attr, ret);
```

Comment:
Shouldn't we validate the value similar to onboard index, `is_valid_onboard_index()`?

Reply:
We are not validating pci hotplug slot number, which also come from ACPI _SUN (except for s390 if I've followed correctly) so I don't think this is needed, but I don't mind adding it either.

Reply:
OK, we can add validation later in case we notice some issues with input data.